contracts/contracts/interfaces/staking-manager.md

```diff
@@ -0,0 +1,109 @@
+- Register a validator on the subnet
+- Set weight of a validator
+- Uptime tracking
+- Remove a validator
+
+Validator/Staking manager contract does not directly operate on P-chain. It sends warp messages and receive warp messages from P-Chain. This ultimately makes it asynchronous with P-Chain, all operations should be considered not final until a related configmation warp message is relayed from P-chain. The manager contract should register it's address on P-chain to be able to interact with the P-chain. The P-chain will only accept warp messages from the registered contract address.
+
+## Register Manager Contract
+
+A Manager contract is expected to be registered in P-Chain, before using any staking operations. This contract will be used to manage the staking operations for the Subnet.
+
+Related ACP section can be found [here.](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets#setsubnetvalidatormanagertx)
+
+A contract must be deployed on the chain and an address must be obtained. This address will be used to register the validator manager contract for the subnet. For the first time only, the `SetSubnetValidatorManagerTx` will be originated from subnet validators (without a sourceAddress in warp message). A response from P-Chain that the contract is registered should be relayed back to the contract, so that the contract can activate itself (start staking, rewarding etc).
+
+Subsequent `SetSubnetValidatorManagerTx` can be originated from the contract to replace the current manager (itself) with another contract. In this case the received confirmation of `SetSubnetValidatorManagerTx` should be relayed back to both contracts, so that the old manager contract can deactivate itself (stop staking, rewarding etc) and the new manager can activate itself.
+
+## Register Validator
+
+Related ACP-77 section [here](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets#step-1-retrieve-a-bls-multisig-from-the-subnet)
+
+Register validator is the process of adding a validator to the subnet.The contract initiates this with a `registerValidator` function that takes following inputs:
+
+- Subnet ID (bytes32): The ID of the subnet where the validator is being registered
+- Node ID (bytes32): The ID of the node that is being registered as a validator
+- Amount (uint64): The staked amount that will determine the weight of the validator
+- expirationTime (uint64): The time when this register message will be expired
+- ed25519 signature (bytes): The signature of the message signed by the node's BLS key
+
+The messageID of the request can be crafted with `sha256(subnetID, nodeID, amount, expirationTime, signature)`. This should be same with the one that will be crafted for P-chain.
+
+The contract should verify these:
+
+- (Optional) the staking amount is greater than the minimum staking amount
+- (Optional) The node is not already registered as a validator. This will be handled by the P-Chain though.
```

Comment:
This will be handled by P-Chain anyway, but I think we have the info to check this in contract